include/mbgl/actor/scheduler.hpp

```diff
@@ -34,6 +39,24 @@ class Scheduler {
     virtual ~Scheduler() = default;
     virtual void schedule(std::weak_ptr<Mailbox>) = 0;
     
+    /*
+     A token to the scheduled action, at the end of
+     the token's life-time, the scheduled action is
+     canceled.
+    */
+    class Scheduled : private util::noncopyable {
+    public:
+        virtual ~Scheduled() = default;
+        
+        // Explicitly cancel the scheduled action
+        virtual void cancel() = 0;
+        
+        virtual bool isFinished() = 0;
+    };
```

Comment:
I've made it extend from `AsyncRequest`. However, we need a way to see if the task is completed (for `Throttler`) that's why I made it a new class with a `isFinished()` method.